src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -134,5 +127,268 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
+                        Debug.Assert(currentPart.ConstantValue is { IsString: true });
+                        baseStringLength += currentPart.ConstantValue.RopeValue!.Length;
+                    }
+
+                    partsBuilder.Add(newPart);
+
+                    ReportDiagnosticsIfObsolete(diagnostics, currentFormatCall.Method, currentPart.Syntax, hasBaseReceiver: false);
+                    ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, currentFormatCall.Method, currentPart.Syntax.Location, isDelegateConversion: false);
+                }
+
+                conversionDiagnostics?.Free();
+
+                var newParts = partsBuilder.ToImmutableAndFree();
+
+                var createMethod = (MethodSymbol)GetWellKnownTypeMember(
+                    WellKnownMember.System_Runtime_CompilerServices_InterpolatedStringBuilder__CreateInt32Int32,
+                    diagnostics,
+                    syntax: unconvertedInterpolatedString.Syntax);
+
+                MethodArgumentInfo? constructorInfo = null;
+
+                if (createMethod != null)
+                {
+                    var intType = GetSpecialType(SpecialType.System_Int32, diagnostics, unconvertedInterpolatedString.Syntax);
+                    var arguments = ImmutableArray.Create<BoundExpression>(
+                        new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(baseStringLength), intType) { WasCompilerGenerated = true },
+                        new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(numFormatHoles), intType) { WasCompilerGenerated = true });
+
+                    constructorInfo = new MethodArgumentInfo(createMethod, arguments, ArgsToParamsOpt: default, DefaultArguments: BitVector.Empty, Expanded: false);
+
+                    ReportDiagnosticsIfObsolete(diagnostics, createMethod, unconvertedInterpolatedString.Syntax, hasBaseReceiver: false);
+                    ReportDiagnosticsIfUnmanagedCallersOnly(diagnostics, createMethod, unconvertedInterpolatedString.Syntax.Location, isDelegateConversion: false);
+                }
+
+                // PROTOTYPE(interp-strings): Do we need to support versions of InterpolatedStringBuilder that do not have Dispose, or implement it
+                // with a different signature, or that are not a ref struct and explicitly implement IDisposable?
+                // PROTOTYPE(interp-string): Is the runtime going to expose this, or do we want to support pattern-based dispose?
+
+                var disposeMethod = (MethodSymbol)GetWellKnownTypeMember(
+                    WellKnownMember.System_Runtime_CompilerServices_InterpolatedStringBuilder__Dispose,
+                    diagnostics,
+                    syntax: unconvertedInterpolatedString.Syntax);
+
+                MethodArgumentInfo? disposeInfo = null;
+
+                if (disposeMethod != null)
```

Comment:
>disposeMethod [](start = 20, length = 13)

Same comments as for constructorInfo

Reply:
No longer ncessary.

---
In reply to: [610916036](https://github.com/dotnet/roslyn/pull/52422#discussion_r610916036) [](ancestors = 610916036)